core/src/main/scala/com/metamx/tranquility/tranquilizer/SimpleTranquilizerAdapter.scala

```diff
@@ -0,0 +1,148 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.metamx.tranquility.tranquilizer
+
+import com.metamx.common.lifecycle.LifecycleStart
+import com.metamx.common.lifecycle.LifecycleStop
+import com.metamx.common.scala.Logging
+import com.twitter.util.Return
+import com.twitter.util.Throw
+import java.io.Closeable
+import java.util.concurrent.atomic.AtomicLong
+import java.util.concurrent.atomic.AtomicReference
+
+/**
+  * Wrap a Tranquilizer, exposing an API that is meant to be easy to use for a single caller that doesn't care
+  * exactly what happens with individual messages, but does care about having exceptions reported eventually.
+  *
+  * Exceptions are reported from time to time when you call "send", and always when you call "flush". Only one
+  * exception will be reported at a time. Note that exceptions triggered by "send" may not necessarily be related to
+  * the actual message you passed; they could be from a previous message. Once an exception is reported, it is cleared
+  * and you may continue sending if you want.
+  *
+  * Calling "start" or "stop" on this adapter will start or stop the underlying Tranquilizer. If you want to start
+  * or stop it yourself, then do not call "start" or "stop" on this adapter.
+  */
+class SimpleTranquilizerAdapter[MessageType] private(
+  tranquilizer: Tranquilizer[MessageType],
+  reportDropsAsExceptions: Boolean
+) extends Logging
+{
+  // failure tracking; need to report failure eventually as we do not return futures from send
+  private val exception = new AtomicReference[Throwable]
+
+  // pending message tracking, for flush
+  private val pending = new AtomicLong
+
+  // counters for metrics
+  private val received = new AtomicLong
+  private val sent     = new AtomicLong
+  private val failed   = new AtomicLong
+
+  /**
+    * Send a message. The actual send may happen asynchronously. If you must know whether or not there was an
+    * exception sending it, call "flush", which will flush all pending messages and throw an exception if there
+    * was one.
+    *
+    * This method may throw an exception if there was one, although it is not guaranteed. Furthermore, if this
+    * method does throw an exception, it might be for a different message.
+    */
```

Comment:
this seems confusing, I wonder how would someone handle the exceptions from this message ? 
e.g if there is an exception  
   -  do we retry sending the same message again as the exception can be from prev message ? 
   -  what should we do in case retry also fails with exception from some other prev message ? 
   -  is it ok to miss exceptions when there are multiple message send failures between two send calls ? 
  
I wonder if it would be less confusing to accept messages async always without any exception and always report exceptions in flush method ?